client/rest-high-level/src/test/java/org/elasticsearch/client/BulkProcessorRetryIT.java

```diff
@@ -0,0 +1,221 @@
+                internalPolicy.logResponse(response);
+                responses.add(response);
+                latch.countDown();
+            }
+
+            @Override
+            public void afterBulk(long executionId, BulkRequest request, Throwable failure) {
+                responses.add(failure);
+                latch.countDown();
+            }
+        }).setBulkActions(1)
+            .setConcurrentRequests(randomIntBetween(0, 100))
+            .setBackoffPolicy(internalPolicy)
+            .build();
+
+        MultiGetRequest multiGetRequest = indexDocs(bulkProcessor, numberOfAsyncOps);
+        latch.await(10, TimeUnit.SECONDS);
+        bulkProcessor.close();
+
+        assertEquals(responses.size(), numberOfAsyncOps);
+
+        boolean rejectedAfterAllRetries = false;
+        for (Object response : responses) {
+            if (response instanceof BulkResponse) {
+                BulkResponse bulkResponse = (BulkResponse) response;
+                for (BulkItemResponse bulkItemResponse : bulkResponse.getItems()) {
+                    if (bulkItemResponse.isFailed()) {
+                        BulkItemResponse.Failure failure = bulkItemResponse.getFailure();
+                        if (failure.getStatus() == RestStatus.TOO_MANY_REQUESTS) {
+                            if (!rejectedExecutionExpected) {
+                                Throwable rootCause = ExceptionsHelper.unwrapCause(failure.getCause());
+                                Iterator<TimeValue> backoffState = internalPolicy.backoffStateFor(bulkResponse);
+                                assertNotNull("backoffState is null (indicates a bulk request got rejected without retry)", backoffState);
+                                if (backoffState.hasNext()) {
+                                    // we're not expecting that we overwhelmed it even once when we maxed out the number of retries
+                                    throw new AssertionError("Got rejected although backoff policy would allow more retries", rootCause);
+                                } else {
+                                    rejectedAfterAllRetries = true;
+                                    logger.debug("We maxed out the number of bulk retries and got rejected (this is ok).");
+                                }
+                            }
+                        } else {
+                            throw new AssertionError("Unexpected failure with status: " + failure.getStatus());
+                        }
+                    }
+                }
+            } else {
+                Throwable t = (Throwable) response;
+                // we're not expecting any other errors
+                throw new AssertionError("Unexpected failure", t);
+            }
+        }
+
+        highLevelClient().indices().refresh(new RefreshRequest());
+        int searchResultCount = highLevelClient().multiGet(multiGetRequest).getResponses().length;
```

Comment:
can you rename this variable, this is not about search anymore, rather the result of multi_get